ddtrace/gitmetadata.py

```diff
@@ -0,0 +1,88 @@
+import os
+import inspect
+import subprocess
+import pkg_resources
+
+
+def __query_git(args):
+    try:
+        p = subprocess.Popen(["git"] + args, stdout=subprocess.PIPE)
+    except EnvironmentError:
+        print("Couldn't run git")
+        return
+    ver = p.communicate()[0]
+    return ver.strip().decode("utf-8")
+
+
+def get_commit_sha():
+    return __query_git(["rev-parse", "HEAD"])
+
+
+def get_repository_url():
+    return __query_git(["config", "--get", "remote.origin.url"])
+
+
+def add_project_urls(setup):
+    source_code_link = get_repository_url() + "#" + get_commit_sha()
+
+    def patch(**attrs):
+        if "project_urls" not in attrs:
+            attrs["project_urls"] = {}
+        attrs["project_urls"]["source_code_link"] = source_code_link
+        return setup(**attrs)
+
+    return patch
+
+
+def __find_package():
+    frm = inspect.stack()[-1]
+    module = inspect.getmodule(frm[0])
```

Comment:
this is probably going to be error prone. I am concerned about our abilities to properly get the user's package name.

have we tested this code with `ddtrace-run python my_script.py`? if I understand this correctly, it would mean it would get our `sitecustomize.py` in this case and then resolve to our `ddtrace` module ?


I don't love the idea... but an alternative is we can scan all available packages and get a list of ones which have the extra urls metadata you are adding.

We already iterate over all packages anyways so we should be able to capture that. The one downside is doing that is expensive, and blocking startup for it is probably not what we want (`pkg_resources` is soooo slow to import... seriously... really slow to just import)